src/status_im2/navigation/events.cljs

```diff
@@ -56,18 +66,42 @@
     {:db                  db
      :navigate-replace-fx go-to-view-id}))
 
-(rf/defn open-modal
-  {:events [:open-modal]}
-  [{:keys [db]} comp screen-params]
-  {:db            (-> (assoc db :view-id comp)
-                      (all-screens-params comp screen-params))
-   :open-modal-fx comp})
-
-(rf/defn init-root
-  {:events [:init-root]}
-  [_ root-id]
-  {:init-root-fx root-id})
+(rf/defn hide-bottom-sheet
+  {:events [:hide-bottom-sheet]}
+  [{:keys [db]}]
+  (let [{:keys [hide? sheets]} (:bottom-sheet db)]
+    (println :hide-bottom-sheet (not hide?) (seq sheets))
+    (when (and (not hide?) (seq sheets))
+      {:db (assoc-in db [:bottom-sheet :hide?] true)})))
 
+(rf/defn bottom-sheet-hidden
+  {:events [:bottom-sheet-hidden]}
+  [{:keys [db]}]
+  (let [{:keys [sheets]} (:bottom-sheet db)
+        rest-sheets      (butlast sheets)]
+    (merge
+     {:db                (assoc db :bottom-sheet {:sheets rest-sheets :hide? false})
+      :hide-bottom-sheet nil
+      :dismiss-keyboard  nil}
+     (when (seq rest-sheets)
+       {:dispatch [:show-next-bottom-sheet]}))))
+
+(rf/defn show-next-bottom-sheet
+  {:events [:show-next-bottom-sheet]}
+  [_]
+  {:show-bottom-sheet nil})
+
+(rf/defn show-bottom-sheet
+  {:events [:show-bottom-sheet]}
+  [{:keys [db]} content]
+  (let [{:keys [sheets hide?]} (:bottom-sheet db)]
+    (rf/merge {:db (update-in db [:bottom-sheet :sheets] #(conj % content))}
+              #(when-not hide?
+                 (if (seq sheets)
+                   (hide-bottom-sheet %)
+                   {:show-bottom-sheet nil})))))
+
+;;LEGACY
```

Comment:
same, for any of this Legacy pieces I would prefer that we rename them so we know they are separate. A comment in a file is not a string enough discriminator imo and it's likely it will go unnoticed for longer this way and more potential to be reused somewhere it should not be.